clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1052,6 +1059,33 @@ wrapSYCLBinariesFromFile(std::vector<module_split::SplitModule> &SplitModules,
   if (!OutputFileOrErr)
     return OutputFileOrErr.takeError();
 
+  StringRef Target = Args.getLastArgValue(OPT_triple_EQ);
+  if (Target.empty())
+    return createStringError(
+        inconvertibleErrorCode(),
+        "can't wrap SYCL image. -triple argument is missed.");
+
+  // SYCL runtime currently works for spir64 target triple and not for
+  // spir64-unknown-unknown/spirv64-unknown-unknown/spirv64.
+  // TODO: Fix SYCL runtime to accept other triples
+  llvm::Triple T(Target);
+  bool isJIT = !(T.getSubArch() == llvm::Triple::SPIRSubArch_gen ||
+                 T.getSubArch() == llvm::Triple::SPIRSubArch_x86_64);
+  SmallString<0> CompileOptions;
```

Comment:
Any reason this is not a `StringRef` instead, given that you are assigning a `StringRef` value in `L1077`?

Reply:
The mentioned `StringRef` refers to a temporary string and the data should be stored in a data-owning container.
In fact, my code contains the bug `use-after-free` and it should be fixed.

Reply:
This code is removed.